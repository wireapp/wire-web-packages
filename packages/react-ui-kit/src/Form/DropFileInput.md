Demo:

```js
import {Columns, Column, DropFileInput} from '@wireapp/react-ui-kit';

<>
  <div>
    <DropFileInput
      onInvalidFilesDropError={console.log}
      onFilesUploaded={() => console.error('error while uploading files')}
      headingText="Drag & Drop an image \nor"
      labelText="select one from your device"
      accept="image/png, image/jpeg"
      description="Image (JPG/PNG) size up to 1 MB, minimum 200 x 600 px"
    />
  </div>
</>;
```
