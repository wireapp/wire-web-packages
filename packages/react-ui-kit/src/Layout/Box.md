Demo:

```js
import {Box} from '@wireapp/react-ui-kit';

<Box>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
  vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
  non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.
</Box>;
```
