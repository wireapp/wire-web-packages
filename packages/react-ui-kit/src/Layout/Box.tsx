/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import React from 'react';
import {COLOR} from '../Identity';

export interface BoxProps<T = HTMLDivElement> extends React.HTMLProps<T> {}

const boxStyles: (props: BoxProps) => ObjectInterpolation<undefined> = props => ({
  border: `2px solid ${COLOR.GRAY_LIGHTEN_72}`,
  borderradius: '8px',
  padding: '16px 32px',
});

const Box = (props: BoxProps) => <div css={boxStyles(props)} {...props} />;

export {Box, boxStyles};
