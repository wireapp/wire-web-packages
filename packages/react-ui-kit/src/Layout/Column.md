Demo:

```js
import {Container, Columns, Column, COLOR} from '@wireapp/react-ui-kit';

const ColumnsStyle = {
  marginBottom: '12px',
};

const ColumnStyle = {
  backgroundColor: COLOR.GRAY_LIGHTEN_32,
  border: `1px solid ${COLOR.GRAY_LIGHTEN_24}`,
};

<Container>
  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
    <Column style={ColumnStyle}>Column</Column>
  </Columns>

  <Columns style={ColumnsStyle}>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
    <Column style={ColumnStyle}>
      <Columns>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
        <Column style={ColumnStyle}>Column</Column>
      </Columns>
    </Column>
  </Columns>
</Container>;
```
