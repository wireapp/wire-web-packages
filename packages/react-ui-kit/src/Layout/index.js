import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export * from './Column';
export * from './Container';
export * from './Content';
export * from './Footer';
export * from './Header';
export * from './StyledApp';
