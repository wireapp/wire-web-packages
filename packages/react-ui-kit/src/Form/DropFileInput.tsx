/*
 * Wire
 * Copyright (C) 2022 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {useState, forwardRef, DragEvent} from 'react';
import * as React from 'react';

import {CSSObject} from '@emotion/react';

import {UploadIcon} from '../Icon';
import {COLOR} from '../Identity';
import {Theme, FlexBox} from '../Layout';
import {TextProps} from '../Text';

export interface DropFileInputProps<T = HTMLInputElement> extends TextProps<T> {
  onFilesUploaded: (files: File[]) => void;
  onInvalidFilesDropError: () => void;
  labelText: string;
  headingText: string;
  description?: string;
  dropFileZoneWrapperCSS?: CSSObject;
}

const visuallyHiddenStyles: CSSObject = {
  clip: 'rect(0 0 0 0)',
  clipPath: 'inset(50%)',
  height: '1px',
  overflow: 'hidden',
  position: 'absolute',
  whiteSpace: 'nowrap',
  width: '1px',
};

export const dropFileWrapperStyle: CSSObject = {
  maxWidth: '330px',
};

export const dropFileZoneWrapperStyle: (theme: Theme, isDraggedOver: boolean) => CSSObject = (
  theme,
  isDraggedOver,
) => ({
  width: '100%',
  padding: '28px 12px',
  border: `1px dashed ${theme.general.primaryColor}`,
  borderBottomWidth: '2px',
  borderRadius: '6px',
  textAlign: 'center',
  backgroundColor: isDraggedOver ? theme.general.backgroundColor : COLOR.WHITE,
  fontWeight: 400,
  fontSize: '0.75rem',
  lineHeight: '13px',
  color: COLOR.GRAY,
});

export const dropFileZoneLabelStyle: (theme: Theme) => CSSObject = theme => ({
  color: theme.general.primaryColor,
  cursor: 'pointer',
  ':focus-within': {
    outline: `1px solid ${theme.general.primaryColor}`,
  },
  ':hover': {
    textDecoration: 'underline',
  },
});

export const dropFileZoneHeadingStyle: CSSObject = {
  display: 'block',
  whiteSpace: 'pre-line',
};

export const dropFileZonDescriptionStyle: CSSObject = {
  marginTop: '12px',
  fontWeight: 400,
  fontSize: '0.625rem',
  lineHeight: '0.8125rem',
  color: COLOR.GRAY,
  whiteSpace: 'pre-line',
};

export const DropFileInput = forwardRef<HTMLInputElement, DropFileInputProps<HTMLInputElement>>(
  (
    {
      onFilesUploaded,
      onInvalidFilesDropError,
      dropFileZoneWrapperCSS,
      labelText,
      headingText,
      description,
      accept,
      multiple,
      ...inputProps
    },
    ref,
  ) => {
    const [isDraggedOver, setIsDraggedOver] = useState(false);

    const resetDraggedOver = () => setIsDraggedOver(false);

    const handleDragOver = (event: DragEvent<HTMLDivElement>) => {
      event.preventDefault();
      event.stopPropagation();
      event.dataTransfer.dropEffect = 'copy';

      if (event.dataTransfer.items.length > 0) {
        setIsDraggedOver(true);
      }
    };

    const handleDrop = (event: DragEvent<HTMLDivElement>) => {
      event.preventDefault();
      event.stopPropagation();

      resetDraggedOver();

      const {files} = event.dataTransfer;
      if (files.length < 1) {
        return;
      }

      const filesArr = multiple ? Array.from(files) : [files[0]];

      const areFilesValid = !!accept
        ? filesArr.every(file =>
            accept
              .split(',')
              .map(v => v.trim())
              .includes(file.type),
          )
        : true;

      if (!areFilesValid) {
        onInvalidFilesDropError();
        return;
      }

      onFilesUploaded(filesArr);
    };

    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      const {files} = event.target;

      if (files.length < 1) {
        return;
      }

      onFilesUploaded(Array.from(files));
    };

    return (
      <div css={dropFileWrapperStyle}>
        <div
          css={(theme: Theme) => ({
            ...dropFileZoneWrapperStyle(theme, isDraggedOver),
            ...dropFileZoneWrapperCSS,
          })}
          data-testid="dropzone"
          onDragOver={handleDragOver}
          onDragLeave={resetDraggedOver}
          onDrop={handleDrop}
        >
          <FlexBox align="center" justify="center" flexWrap="wrap" css={{maxWidth: '280px', margin: '0 auto'}}>
            <UploadIcon css={{margin: '12px 0'}} />
            <div css={{maxWidth: '160px', margin: '0 25px'}}>
              <span css={dropFileZoneHeadingStyle}>{headingText}</span>
              <label
                aria-label={`${headingText} ${labelText} (${description})`}
                css={(theme: Theme) => dropFileZoneLabelStyle(theme)}
              >
                <span>{labelText}</span>
                <input
                  data-testid="file-input"
                  ref={ref}
                  accept={accept}
                  multiple={multiple}
                  css={visuallyHiddenStyles}
                  onChange={handleChange}
                  type="file"
                  {...inputProps}
                />
              </label>
            </div>
          </FlexBox>
        </div>
        {description && <p css={dropFileZonDescriptionStyle}>{description}</p>}
      </div>
    );
  },
);
DropFileInput.displayName = 'DropFileInput';
