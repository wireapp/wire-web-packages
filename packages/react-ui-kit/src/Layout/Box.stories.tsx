/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {Meta, StoryObj} from '@storybook/react';

import {Box} from './Box';

import {COLOR} from '../Identity/colors';
import {Text} from '../Text';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Layout/Box',
  decorators: [
    Story => (
      <div style={{padding: '24px', maxWidth: '600px', margin: '0 auto'}}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Box>;

const LOREM_IPSUM = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.`;

export const Default: Story = {
  args: {
    children: LOREM_IPSUM,
  },
};

export const WithCustomPadding: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {padding: '32px'},
  },
};

export const WithCustomBorder: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {border: `2px solid ${COLOR.BLUE}`},
  },
};

export const WithFormattedContent: Story = {
  args: {
    children: (
      <>
        <Text block style={{fontSize: '20px', fontWeight: 'bold', marginBottom: '16px'}}>
          Important Notice
        </Text>
        <Text block style={{marginBottom: '12px'}}>
          {LOREM_IPSUM}
        </Text>
        <Text block style={{color: COLOR.GRAY}}>
          Last updated: 2025-01-01
        </Text>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Text block style={{marginBottom: '16px'}}>
          Outer box content
        </Text>
        <Box style={{backgroundColor: COLOR.GRAY_LIGHTEN_92}}>
          <Text>Inner box content</Text>
        </Box>
      </>
    ),
  },
};

export const Interactive: Story = {
  args: {
    children: (
      <div style={{textAlign: 'center'}}>
        <Text block style={{marginBottom: '16px'}}>
          Click the button below
        </Text>
        <button
          style={{
            backgroundColor: COLOR.BLUE,
            border: 'none',
            borderRadius: '4px',
            color: COLOR.WHITE,
            cursor: 'pointer',
            padding: '8px 16px',
          }}
        >
          Action Button
        </button>
      </div>
    ),
    style: {cursor: 'pointer'},
  },
};
