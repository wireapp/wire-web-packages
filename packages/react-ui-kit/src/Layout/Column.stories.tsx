/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {ReactNode} from 'react';

import {Meta, StoryObj} from '@storybook/react';

import {Column, Columns} from './Column';

import {COLOR} from '../Identity/colors';
import {QueryKeys} from '../mediaQueries';

const meta: Meta<typeof Columns> = {
  component: Columns,
  title: 'Layout/Column',
};

export default meta;
type Story = StoryObj<typeof Columns>;

const Box = ({children}: {children: ReactNode}) => (
  <div
    style={{
      backgroundColor: COLOR.GRAY_LIGHTEN_72,
      border: `1px solid ${COLOR.GRAY_LIGHTEN_48}`,
      borderRadius: '4px',
      padding: '16px',
      textAlign: 'center',
    }}
  >
    {children}
  </div>
);

export const Default: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Column 1</Box>
        </Column>
        <Column>
          <Box>Column 2</Box>
        </Column>
        <Column>
          <Box>Column 3</Box>
        </Column>
      </>
    ),
  },
};

export const TwoColumns: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Column 1</Box>
        </Column>
        <Column>
          <Box>Column 2</Box>
        </Column>
      </>
    ),
  },
};

export const ResponsiveColumns: Story = {
  args: {
    query: QueryKeys.TABLET,
    children: (
      <>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
        <Column>
          <Box>Stacks on tablet</Box>
        </Column>
      </>
    ),
  },
};

export const NestedColumns: Story = {
  args: {
    children: (
      <>
        <Column>
          <Box>Main Column</Box>
        </Column>
        <Column>
          <Columns>
            <Column>
              <Box>Nested 1</Box>
            </Column>
            <Column>
              <Box>Nested 2</Box>
            </Column>
          </Columns>
        </Column>
      </>
    ),
  },
};
