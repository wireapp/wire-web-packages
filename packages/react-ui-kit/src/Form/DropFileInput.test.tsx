/*
 * Wire
 * Copyright (C) 2022 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import React from 'react';
import {matchComponent} from '../test/testUtil';
import {DropFileInput, DropFileInputProps} from './DropFileInput';
import {fireEvent, render} from '@testing-library/react';
import {StyledApp, THEME_ID} from '../Layout';

//note: we don't have to test file type validation using native file upload,
// it won't allow to pick different type thant expected
interface GetDefaultPropsType {
  accept?: string;
  multiple?: boolean;
}

const getDefaultProps = ({accept, multiple}: GetDefaultPropsType) => ({
  accept,
  multiple,
  onInvalidFilesDropError: jest.fn(),
  onFilesUploaded: jest.fn(),
  headingText: 'Drag & Drop an image \nor',
  labelText: 'select one from your device',
  description: 'Image (JPG/PNG) size up to 1 MB, minimum 200 x 600 px',
});

const ThemedDropFileInput = (props: DropFileInputProps) => (
  <StyledApp themeId={THEME_ID.LIGHT}>
    <DropFileInput {...props} />
  </StyledApp>
);

const pngFile = new File(['(⌐□_□)'], 'chucknorris.png', {type: 'image/png'});
const jpegFile = new File(['(⌐□_□)'], 'chucknorris.jpg', {type: 'image/jpeg'});
const xlsxFile = new File(['(⌐□_□)'], 'chucknorris.xlsx', {type: '.xlsx'});

describe('"DropFileInput"', () => {
  it('matches snapshot', () =>
    matchComponent(<ThemedDropFileInput {...getDefaultProps({accept: 'image/png, image/jpeg'})} />));

  it('returns file on native file upload', () => {
    const props = getDefaultProps({});

    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
    const fileInput = getByTestId('file-input');

    // i have to do this because `input.files =[file]` is not allowed
    Object.defineProperty(fileInput, 'files', {
      value: [pngFile],
    });
    fireEvent.change(fileInput);

    expect(props.onFilesUploaded).toHaveBeenCalledWith([pngFile]);
  });

  it('allows to upload different types of files when "accept" attribute not specified', () => {
    const props = getDefaultProps({});

    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
    const dropZone = getByTestId('dropzone');

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [pngFile],
      },
    });

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [xlsxFile],
      },
    });

    expect(props.onFilesUploaded).toHaveBeenCalledTimes(2);
  });

  it('allows to upload only file types specified in "accept" attribute', () => {
    const props = getDefaultProps({accept: 'image/png'});

    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
    const dropZone = getByTestId('dropzone');

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [pngFile],
      },
    });

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [jpegFile],
      },
    });

    expect(props.onFilesUploaded).toHaveBeenCalledTimes(1);
    expect(props.onInvalidFilesDropError).toHaveBeenCalled();
  });

  it('returns first file when "multiple" attribute not specified', () => {
    const props = getDefaultProps({
      accept: 'image/png, image/jpeg',
    });

    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
    const dropZone = getByTestId('dropzone');

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [pngFile, jpegFile],
      },
    });

    expect(props.onFilesUploaded).toHaveBeenCalledWith([pngFile]);
  });

  it('returns all files when "multiple" attribute specified', () => {
    const props = getDefaultProps({
      accept: 'image/png, image/jpeg',
      multiple: true,
    });

    const {getByTestId} = render(<ThemedDropFileInput {...props} />);
    const dropZone = getByTestId('dropzone');

    fireEvent.drop(dropZone, {
      dataTransfer: {
        files: [pngFile, jpegFile],
      },
    });

    expect(props.onFilesUploaded).toHaveBeenCalledWith([pngFile, jpegFile]);
  });
});
