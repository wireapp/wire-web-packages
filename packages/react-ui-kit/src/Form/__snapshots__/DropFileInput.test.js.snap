// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"DropFileInput" matches snapshot 1`] = `
<div
  className="css-7uml9n-StyledAppContainer"
>
  <div
    className="css-7uml9n-StyledAppContainer"
  >
    <div
      className="css-1y7m5n0"
    >
      <div
        className="css-qtc2fr"
        data-testid="dropzone"
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
      >
        <div
          className="css-c9hw9m-DropFileInput"
        >
          <svg
            className="css-t97fbx-DropFileInput"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <g>
              <path
                d="M0 14H16V16H0V14ZM7 11H9V4H13L8 0L3 4H7V11Z"
              />
            </g>
          </svg>
          <div
            className="css-r5lgab-DropFileInput"
          >
            <span
              className="css-1kr0aza"
            >
              Drag & Drop an image 
or
            </span>
            <label
              aria-label="Drag & Drop an image 
or select one from your device (Image (JPG/PNG) size up to 1 MB, minimum 200 x 600 px)"
              className="css-ylkvue"
            >
              <span>
                select one from your device
              </span>
              <input
                accept="image/png, image/jpeg"
                className="css-5ur8x"
                data-testid="file-input"
                onChange={[Function]}
                type="file"
              />
            </label>
          </div>
        </div>
      </div>
      <p
        className="css-14vo0st"
      >
        Image (JPG/PNG) size up to 1 MB, minimum 200 x 600 px
      </p>
    </div>
  </div>
</div>
`;
